src/main/kotlin/org/opensearch/replication/action/stats/TransportLeaderStatsAction.kt

```diff
@@ -0,0 +1,100 @@
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.replication.action.stats
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.GlobalScope
+import org.apache.logging.log4j.LogManager
+import org.opensearch.action.FailedNodeException
+import org.opensearch.action.support.ActionFilters
+import org.opensearch.action.support.nodes.TransportNodesAction
+import org.opensearch.client.node.NodeClient
+import org.opensearch.cluster.service.ClusterService
+import org.opensearch.common.inject.Inject
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.replication.seqno.RemoteClusterStats
+import org.opensearch.threadpool.ThreadPool
+import org.opensearch.transport.TransportService
+import java.util.concurrent.TimeUnit
+
+class TransportLeaderStatsAction @Inject constructor(transportService: TransportService,
+                                                     clusterService: ClusterService,
+                                                     threadPool: ThreadPool,
+                                                     actionFilters: ActionFilters,
+                                                     private val remoteStats: RemoteClusterStats,
+                                                     private val client: NodeClient) :
+        TransportNodesAction<LeaderStatsRequest, LeaderStatsResponse, NodeStatsRequest, LeaderNodeStatsResponse>(LeaderStatsAction.NAME,
+             threadPool, clusterService, transportService,  actionFilters, ::LeaderStatsRequest,  ::NodeStatsRequest, ThreadPool.Names.MANAGEMENT,
+                LeaderNodeStatsResponse::class.java), CoroutineScope by GlobalScope {
+
+    companion object {
+        private val log = LogManager.getLogger(TransportLeaderStatsAction::class.java)
+        val durationThresholdActiveReplication =TimeUnit.SECONDS.toMillis(300) //5 min
+    }
+
+    override fun newNodeRequest(request: LeaderStatsRequest): NodeStatsRequest {
+       return NodeStatsRequest()
+    }
+
+    override fun newNodeResponse(input: StreamInput): LeaderNodeStatsResponse {
+        return LeaderNodeStatsResponse(input)
+    }
+
+    override fun newResponse(request: LeaderStatsRequest?, responses: MutableList<LeaderNodeStatsResponse>?, failures: MutableList<FailedNodeException>?): LeaderStatsResponse {
+        return LeaderStatsResponse(clusterService.clusterName, responses, failures)
+
+    }
+
+     private fun isIndexReplicationActive(index :String) :Boolean {
+        val indicesRouting = clusterService.state().routingTable.indicesRouting
+         if (!indicesRouting.containsKey(index)) {
+             return false
+         }
+
+         return true
+         // Retention lease is the most accurate method to find out active replication, however it blocks the generic thread pool
+         // given that nodeOperation doesn't support async calls
```

Comment:
Yeah, we should create an issue to track. One option could be to also persist state on leader side on replication starting and cleaning on stopping.

Reply:
Created one : https://github.com/opensearch-project/cross-cluster-replication/issues/124

Reply:
Just saw this - `indexShard` has `getRetentionLeases`  method that can be used to filter the retention leases instead of a transport call. we can discuss